dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry/internal/rest/TelemetryClientFilter.java

```diff
@@ -51,6 +51,19 @@ public class TelemetryClientFilter implements ClientRequestFilter, ClientRespons
     private static final NetClientAttributesGetterImpl NET_CLIENT_ATTRIBUTES_GETTER = new NetClientAttributesGetterImpl();
     private static final HttpClientAttributesGetterImpl HTTP_CLIENT_ATTRIBUTES_GETTER = new HttpClientAttributesGetterImpl();
 
+    /**
+     * Retrieve the TelemetryClientFilter for the current application using CDI
+     * <p>
+     * Implementation note: It's important that there's a class which is registered as a CDI bean on the stack from this bundle when {@code CDI.current()} is called.
+     * <p>
+     * Doing this in a static method within {@code TelemetryClientFilter} accomplishes this.
```

Comment:
Accomplishes what? The way it reads to me suggest that using a static method here accomplishes the goal of having a CDI bean on the stack ... but that isn't right?

Reply:
That is right, `TelemetryClientFilter` is the bean class on the stack. It's the bean we're looking up with the `CDI.current().select()` call.

Reply:
Oh, it artificially forces TelemetryClientFilter onto the stack so that the `CDI.current().select()` call always finds the right thing?

Reply:
Yes :+1: